napari/layers/_layer_actions.py

```diff
@@ -53,16 +53,18 @@ def _convert(ll: LayerList, type_: str) -> None:
 
     for lay in list(ll.selection):
         idx = ll.index(lay)
-        ll.pop(idx)
 
         if isinstance(lay, Shapes) and type_ == 'labels':
             data = lay.to_labels()
+            idx += 1
         elif (
             not np.issubdtype(lay.data.dtype, np.integer) and type_ == 'labels'
         ):
             data = lay.data.astype(int)
```

Comment:
@Czaki this type conversion will result in a copy of the data right?
So for float images, could add the labels instead of pop/replace?

Reply:
yes